java-generator/core/src/main/java/io/fabric8/java/generator/Config.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.java.generator;
+
+public class Config {
+    private boolean uppercaseEnums;
+
+    public Config(boolean uppercaseEnums) {
+        this.uppercaseEnums = uppercaseEnums;
+    }
+
+    public boolean isUppercaseEnums() {
+        return uppercaseEnums;
+    }
+
+    public void setUppercaseEnums(boolean uppercaseEnums) {
+        this.uppercaseEnums = uppercaseEnums;
+    }
+}
```

Comment:
Note for next iteration.

I think this is aligned with what we discussed.
However, now we are providing the defaults via the CLI and the plugin. 

This will create potential issues in the future, since the default behavior of the tool might differ depending on the way it's invoked.

The default value should either be provided by the Config class itself:
e.g. 
```java
    public boolean isUppercaseEnums() {
        return uppercaseEnums == null ? $default : upperCaseEnums;
    }
```

Or at the consumer service (in case this is only consumed in a single place). I'd prefer the former approach.

This way we guarantee that defaults are managed at a single place, and we could even provide tests that guaranteed the "default behavior spec".

Reply:
Also, I'm not sure you defaulted to `false` since that provides code that doesn't look right from a Java programmer's PoV.

Reply:
Regarding the first point a possibility is to have `static` fields in the `Config` class:

```java
public final static boolean defaultUppercaseEnum = maybe;
```
and consume this source of truth in the relevant annotations in the Maven plugin and the CLI.

Reply:
Regarding defaulting to `false`, I don't have super strong opinions in this regard.

Reply:
@manusa followed up:
https://github.com/fabric8io/kubernetes-client/pull/3812/commits/3608fdbebf421245b95fffea91d82de0c3c01a05

Reply:
My comment was intended for a following PR :sweat_smile: 

> and consume this source of truth in the relevant annotations in the Maven plugin and the CLI.

For me this is still something that spreads the logic around without need. Why would we want to expose static variables when all the logic could be self contained within the Config class?

IMHO Config should be:
- Immutable. You can only clone it or create it, there are no setters.
- Since we have SundrIO, a public builder should be provided.
- Getters are public and contain all the logic.
- No default computing logic is exposed

Following this criteria the Maven Plugin would:
```java
/* ... */
  @Option(required = false)
  Boolean uppercaseEnum;

  @Option(required = false)
  String paramOther;
/* ... */
public void run() {
  new ConfigBuilder().withUppercaseEnum(uppercaseEnum).withParamOther(paramOther)...build();
/* ... */
```

As stated, the defaults are centralized in the Config class:

```java
private static final boolean DEFAULT_UPPERCASE_ENUM = true;
private static final boolean DEFAULT_PARAM_OTHER = "default";
/* ... */
isUppercaseEnum() {
  return getOrDefault(uppercaseEnum, DEFAULT_UPPERCASE_ENUM);
}
getParamOther() {
  return getOrDefault(paramOther, DEFAULT_PARAM_OTHER);
}
/* ... */
```

The logic is self-contained and not exposed, neither the plugins, the CLI, or anything else needs to know about defaults. They should only deal with configuration.

If at some point defaults need to be changed, only Config and the Config tests need to be updated. CLI, plugins, etc. are isolated from this.

Reply:
@manusa I think I have done everything but the `sundrio` integration that resulted to be more tricky than expected.